When `aqua.yaml` does not contain `packages` field, `aqua g -i` errors with `SIGSEGV`

### aqua info

```console
$ aqua info
{
  "version": "2.28.0",
  "commit_hash": "ecd9e704110bf2f236b744cd6497ad50e1b52e47",
  "os": "linux",
  "arch": "amd64",
  "pwd": "/home/(USER)",
  "root_dir": "/home/(USER)/.local/share/aquaproj-aqua",
  "env": {
    "AQUA_GLOBAL_CONFIG": "/home/(USER)/.config/aquaproj-aqua/aqua.yaml:",
    "AQUA_POLICY_CONFIG": "/home/(USER)/.config/aquaproj-aqua/policy.yaml:"
  },
  "config_files": []
}
```


### Overview

When `aqua.yaml` does not contain `packages` field, `aqua g -i` errors with `SIGSEGV`.

### How to reproduce

aqua.yaml

```yaml
registries:
  - type: standard
    ref: v4.182.0
```

Executed command and output

```console
$ mkdir foo
$ cd foo
$ nano aqua.yaml # copy the aqua.yaml above
$ aqua g -i aquaproj/example-go-slsa-provenance
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xac9fc1]

goroutine 1 [running]:
github.com/aquaproj/aqua/v2/pkg/controller/generate/output.appendPkgsNode(0x0, {0xed3140, 0xc000092900})
        /home/runner/work/aqua/aqua/pkg/controller/generate/output/insert.go:48 +0x21
github.com/aquaproj/aqua/v2/pkg/controller/generate/output.updateASTFile({0xed30c0, 0xc000d91300}, {0xc0014c6208?, 0x1?, 0xc28b40?})
        /home/runner/work/aqua/aqua/pkg/controller/generate/output/insert.go:73 +0xfa
github.com/aquaproj/aqua/v2/pkg/controller/generate/output.(*Outputter).generateInsert(0xc000091e40, {0xc0000281a0, 0x19}, {0xc0014c6208, 0x1, 0x1})
        /home/runner/work/aqua/aqua/pkg/controller/generate/output/insert.go:33 +0x230
github.com/aquaproj/aqua/v2/pkg/controller/generate/output.(*Outputter).Output(0xc0004335e0?, 0xc000d90a40)
        /home/runner/work/aqua/aqua/pkg/controller/generate/output/output.go:41 +0x1f5
github.com/aquaproj/aqua/v2/pkg/controller/generate.(*Controller).Generate(0xc0000ba360, {0xece700, 0xc000139d00}, 0xc0004335e0, 0xc0000b8580, {0xc000035b60, 0x1, 0x1})
        /home/runner/work/aqua/aqua/pkg/controller/generate/generate.go:56 +0x20e
github.com/aquaproj/aqua/v2/pkg/cli.(*Runner).generateAction(0xc0001890e0, 0xc000022440)
        /home/runner/work/aqua/aqua/pkg/cli/generate.go:189 +0x29c
github.com/urfave/cli/v2.(*Command).Run(0xc0000b4c60, 0xc000022440, {0xc0000b6480, 0x3, 0x3})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:276 +0x97d
github.com/urfave/cli/v2.(*Command).Run(0xc0000b82c0, 0xc000022140, {0xc000138040, 0x4, 0x4})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:269 +0xbb7
github.com/urfave/cli/v2.(*App).RunContext(0xc0000b0000, {0xece700, 0xc000139d00}, {0xc000138040, 0x4, 0x4})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/app.go:333 +0x58b
github.com/aquaproj/aqua/v2/pkg/cli.(*Runner).Run(0xc0001890e0, {0xece700, 0xc000139d00}, {0xc000138040, 0x4, 0x4})
        /home/runner/work/aqua/aqua/pkg/cli/runner.go:221 +0x2b45
main.core(0xc0004335e0, 0xc000149680)
        /home/runner/work/aqua/aqua/cmd/aqua/main.go:56 +0x25a
main.main()
        /home/runner/work/aqua/aqua/cmd/aqua/main.go:30 +0x4c
```


### Debug output

_No response_

### Expected behaviour

Either

- No error happens
- An error that tells `packages` field is required
  - also it'd be nice to have a message that `aqua init` exists


### Actual behaviour

SIGSEGV error happens.

### Note

_No response_

--- a/pkg/controller/generate/output/insert.go
+++ b/pkg/controller/generate/output/insert.go
@@ -104,33 +104,5 @@ func updateASTFile(body ast.Node, pkgs []*aqua.Package) error {
 		return fmt.Errorf(`find a mapping value node "packages": %w`, err)
 	}
 
-	if values == nil {
-		values, err := wast.NormalizeMappingValueNodes(body)
-		if err != nil {
-			return err
-		}
-		idx := len(values)
-		mn, ok := node.(*ast.MappingValueNode)
-		if !ok {
-			return errors.New("body must be a mapping value node")
-		}
-		mv, ok := body.(*ast.MappingNode)
-		if !ok {
-			return errors.New("body must be a mapping node")
-		}
-		latterValues := make([]*ast.MappingValueNode, len(mv.Values[idx:]))
-		copy(latterValues, mv.Values[idx:])
-		mv.Values = mv.Values[:idx]
-		mv.Merge(&ast.MappingNode{
-			Values: []*ast.MappingValueNode{
-				mn,
-			},
-		})
-		mv.Merge(&ast.MappingNode{
-			Values: latterValues,
-		})
-		return nil
-	}
-
 	return appendPkgsNode(values, node)
 }

--- a/pkg/controller/generate/output/insert.go
+++ b/pkg/controller/generate/output/insert.go
@@ -30,41 +30,8 @@ func (o *Outputter) generateInsert(cfgFilePath string, pkgs []*aqua.Package) err
 			"num_of_docs": len(file.Docs),
 		})
 	}
-	body := file.Docs[0].Body
-
-	values, err := wast.FindMappingValueFromNode(body, "packages")
+	s, err := o.getAppendedTxt(cfgFilePath, file, pkgs)
 	if err != nil {
-		return fmt.Errorf(`find a mapping value node "packages": %w`, err)
-	}
-	if values == nil {
-		a, err := yaml.Marshal(struct {
-			Packages []*aqua.Package `yaml:"packages"`
-		}{
-			Packages: pkgs,
-		})
-		if err != nil {
-			return fmt.Errorf("marshal packages: %w", err)
-		}
-		b, err := afero.ReadFile(o.fs, cfgFilePath)
-		if err != nil {
-			return fmt.Errorf("read a configuration file: %w", err)
-		}
-		sb := string(b)
-		if !strings.HasSuffix(sb, "\n") {
-			sb += "\n"
-		}
-		sb += string(a)
-		stat, err := o.fs.Stat(cfgFilePath)
-		if err != nil {
-			return fmt.Errorf("get configuration file stat: %w", err)
-		}
-		if err := afero.WriteFile(o.fs, cfgFilePath, []byte(sb), stat.Mode()); err != nil {
-			return fmt.Errorf("write the configuration file: %w", err)
-		}
-		return nil
-	}
-
-	if err := updateASTFile(body, pkgs); err != nil {
 		return err
 	}
 
@@ -72,10 +39,26 @@ func (o *Outputter) generateInsert(cfgFilePath string, pkgs []*aqua.Package) err
 	if err != nil {
 		return fmt.Errorf("get configuration file stat: %w", err)
 	}
-	if err := afero.WriteFile(o.fs, cfgFilePath, []byte(file.String()), stat.Mode()); err != nil {
+	if err := afero.WriteFile(o.fs, cfgFilePath, []byte(s), stat.Mode()); err != nil {
 		return fmt.Errorf("write the configuration file: %w", err)
 	}
 	return nil
+}
+
+func (o *Outputter) getAppendedTxt(cfgFilePath string, file *ast.File, pkgs []*aqua.Package) (string, error) {
+	body := file.Docs[0].Body
+	values, err := wast.FindMappingValueFromNode(body, "packages")
+	if err != nil {
+		return "", fmt.Errorf(`find a mapping value node "packages": %w`, err)
+	}
+	if values == nil {
+		return o.appendPkgsTxt(cfgFilePath, pkgs)
+	}
+
+	if err := updateASTFile(values, pkgs); err != nil {
+		return "", err
+	}
+	return file.String(), nil
 }
 
 func appendPkgsNode(mapValue *ast.MappingValueNode, node ast.Node) error {
@@ -93,16 +76,31 @@ func appendPkgsNode(mapValue *ast.MappingValueNode, node ast.Node) error {
 	}
 }
 
-func updateASTFile(body ast.Node, pkgs []*aqua.Package) error {
+func updateASTFile(values *ast.MappingValueNode, pkgs []*aqua.Package) error {
 	node, err := yaml.ValueToNode(pkgs)
 	if err != nil {
 		return fmt.Errorf("convert packages to node: %w", err)
 	}
 
-	values, err := wast.FindMappingValueFromNode(body, "packages")
-	if err != nil {
-		return fmt.Errorf(`find a mapping value node "packages": %w`, err)
-	}
-
 	return appendPkgsNode(values, node)
+}
+
+func (o *Outputter) appendPkgsTxt(cfgFilePath string, pkgs []*aqua.Package) (string, error) {
+	a, err := yaml.Marshal(struct {
+		Packages []*aqua.Package `yaml:"packages"`
+	}{
+		Packages: pkgs,
+	})
+	if err != nil {
+		return "", fmt.Errorf("marshal packages: %w", err)
+	}
+	b, err := afero.ReadFile(o.fs, cfgFilePath)
+	if err != nil {
+		return "", fmt.Errorf("read a configuration file: %w", err)
+	}
+	sb := string(b)
+	if !strings.HasSuffix(sb, "\n") {
+		sb += "\n"
+	}
+	return sb + string(a), nil
 }

--- a/pkg/controller/generate/output/insert.go
+++ b/pkg/controller/generate/output/insert.go
@@ -3,6 +3,7 @@ package output
 import (
 	"errors"
 	"fmt"
+	"strings"
 
 	wast "github.com/aquaproj/aqua/v2/pkg/ast"
 	"github.com/aquaproj/aqua/v2/pkg/config/aqua"
@@ -29,8 +30,41 @@ func (o *Outputter) generateInsert(cfgFilePath string, pkgs []*aqua.Package) err
 			"num_of_docs": len(file.Docs),
 		})
 	}
+	body := file.Docs[0].Body
 
-	if err := updateASTFile(file.Docs[0].Body, pkgs); err != nil {
+	values, err := wast.FindMappingValueFromNode(body, "packages")
+	if err != nil {
+		return fmt.Errorf(`find a mapping value node "packages": %w`, err)
+	}
+	if values == nil {
+		a, err := yaml.Marshal(struct {
+			Packages []*aqua.Package `yaml:"packages"`
+		}{
+			Packages: pkgs,
+		})
+		if err != nil {
+			return fmt.Errorf("marshal packages: %w", err)
+		}
+		b, err := afero.ReadFile(o.fs, cfgFilePath)
+		if err != nil {
+			return fmt.Errorf("read a configuration file: %w", err)
+		}
+		sb := string(b)
+		if !strings.HasSuffix(sb, "\n") {
+			sb += "\n"
+		}
+		sb += string(a)
+		stat, err := o.fs.Stat(cfgFilePath)
+		if err != nil {
+			return fmt.Errorf("get configuration file stat: %w", err)
+		}
+		if err := afero.WriteFile(o.fs, cfgFilePath, []byte(sb), stat.Mode()); err != nil {
+			return fmt.Errorf("write the configuration file: %w", err)
+		}
+		return nil
+	}
+
+	if err := updateASTFile(body, pkgs); err != nil {
 		return err
 	}
 
@@ -68,6 +102,34 @@ func updateASTFile(body ast.Node, pkgs []*aqua.Package) error {
 	values, err := wast.FindMappingValueFromNode(body, "packages")
 	if err != nil {
 		return fmt.Errorf(`find a mapping value node "packages": %w`, err)
+	}
+
+	if values == nil {
+		values, err := wast.NormalizeMappingValueNodes(body)
+		if err != nil {
+			return err
+		}
+		idx := len(values)
+		mn, ok := node.(*ast.MappingValueNode)
+		if !ok {
+			return errors.New("body must be a mapping value node")
+		}
+		mv, ok := body.(*ast.MappingNode)
+		if !ok {
+			return errors.New("body must be a mapping node")
+		}
+		latterValues := make([]*ast.MappingValueNode, len(mv.Values[idx:]))
+		copy(latterValues, mv.Values[idx:])
+		mv.Values = mv.Values[:idx]
+		mv.Merge(&ast.MappingNode{
+			Values: []*ast.MappingValueNode{
+				mn,
+			},
+		})
+		mv.Merge(&ast.MappingNode{
+			Values: latterValues,
+		})
+		return nil
 	}
 
 	return appendPkgsNode(values, node)

